### PR TITLE
B-18326 Update the Estimated Weight - B-18327 Remove Fields - B-18322 Update Requested pickup date

### PR DESCRIPTION
**Pull request by WeatherfordAaron**
Created from [old PR](https://github.com/transcom/mymove/pull/11757)

### AC copied from the Agility Tickets
[B-18326](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3a873997): Given that I am a TOO, when reviewing the PPM dashboard, then I will see "Estimated Weight" which displays the customer's estimated weight they entered during onboarding.

[B-18327](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3a874041): Given that I am a TOO, when reviewing the PPM dashboard, then I will not see fields "requested delivery date", "Scheduled delivery date",  "Actual delivery date", and "Required Delivery Date"

[B-18322](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3a873997): Given that I am a TOO, when reviewing the PPM dashboard, then I will see requested pick up date adjusted to planned move date which displays the customer's planned move date entered during onboarding

### How to test

1. Make a PPM and note what is entered for the estimated weight during shipment creation.
2. Open the PPM shipment on the MTO tab as a TOO user.
3. Observe that the estimated weight entered matches that entered by the customer.
4. Observe that none of the follow fields show up:
    -Requested Delivery Date
    -Scheduled Delivery Date
    -Actual Delivery Date
    -Required Delivery Date
    -Requested Pickup Date
    -Scheduled Pickup Date
    -Actual Pickup Date
5. Observe that the following fields and values show up in stead:
    -Planned Move Date
    -Actual Move Date
    -Actual Arrival Date

## Screenshots
Screenshot 2024-01-16 at 6 09 17 PM

### CODE REVIEW

**Comment by JamesHawks224** [comment]
Your screenshot has the old 3 fields.  Suggest replacing w/expected result.

File: src/components/Office/ImportantShipmentDates/ImportantShipmentDates.jsx [hunk 1]
```diff
@@ -15,20 +15,27 @@ const ImportantShipmentDates = ({
...
   return (
     <div className={classnames('maxw-tablet', styles.shipmentDatesContainer)}>
       <DataTableWrapper className="table--data-point-group">
-        <DataTable columnHeaders={['Required Delivery Date']} dataRow={[requiredDeliveryDate || emDash]} />
+        {!isPPM && <DataTable columnHeaders={['Required Delivery Date']} dataRow={[requiredDeliveryDate || emDash]} />}
         <DataTable
-          columnHeaders={['Requested pick up date', 'Scheduled pick up date', 'Actual pick up date']}
+          columnHeaders={[headerPlannedMoveDate, headerActualMoveDate, headerActualArrivalDate]}
           dataRow={[requestedPickupDate || emDash, scheduledPickupDate || emDash, actualPickupDate || emDash]}
         />
```
Comment (JamesHawks224) [inline src/components/Office/ImportantShipmentDates/ImportantShipmentDates.jsx:32]: Header labels are consistent if move is PPM, but data var names are consistent with HHG.  Suggest both header and labels be consistent w/each other i.e. if HHG header name vars, then HHG data name vars and vice versa if choosing PPM var names.

File: src/components/Office/ShipmentDetails/ShipmentDetailsMain.jsx [hunk 2]
```diff
@@ -124,23 +125,52 @@ const ShipmentDetailsMain = ({
...
+      displayedPickupAddress = pickupAddress;
+      displayedDeliveryAddress = destinationAddress || destinationDutyLocationAddress;
+      break;
     default:
+      pickupRequestedDate = requestedPickupDate;
+      pickupScheduledDate = scheduledPickupDate;
+      pickupActualDate = actualPickupDate;
+      weightResult = primeEstimatedWeight;
       displayedPickupAddress = pickupAddress;
       displayedDeliveryAddress = destinationAddress || destinationDutyLocationAddress;
```
Comment (JamesHawks224) [inline src/components/Office/ShipmentDetails/ShipmentDetailsMain.jsx:172]: There fields added for multiple shipment types.  Test steps  has a test for PPM but not the other types.  Should they be added to the testing steps?

File: src/components/Office/ShipmentDetails/ShipmentDetailsMain.jsx [hunk 3]
```diff
@@ -183,13 +213,14 @@ const ShipmentDetailsMain = ({
...
+        scheduledPickupDate={scheduledPickupDate ? formatDateWithUTC(pickupScheduledDate) : null}
+        actualPickupDate={actualPickupDate ? formatDateWithUTC(pickupActualDate) : null}
         requestedDeliveryDate={requestedDeliveryDate ? formatDateWithUTC(requestedDeliveryDate) : null}
         scheduledDeliveryDate={scheduledDeliveryDate ? formatDateWithUTC(scheduledDeliveryDate) : null}
         actualDeliveryDate={actualDeliveryDate ? formatDateWithUTC(actualDeliveryDate) : null}
         requiredDeliveryDate={requiredDeliveryDate ? formatDateWithUTC(requiredDeliveryDate) : null}
+        isPPM={shipmentType === SHIPMENT_OPTIONS.PPM}
       />
       <ShipmentAddresses
         pickupAddress={displayedPickupAddress}
```
Comment (JamesHawks224) [inline src/components/Office/ShipmentDetails/ShipmentDetailsMain.jsx:226]: There are fields added for multiple shipment types.  There's a test for PPM but not the other types.  Should they be added?

File: src/components/Office/ImportantShipmentDates/ImportantShipmentDates.jsx (hunk 1 above)
Comment (WeatherfordAaron) [reply src/components/Office/ImportantShipmentDates/ImportantShipmentDates.jsx:32]: Good callout. I'm going to add some logic to resolve this.

File: src/components/Office/ShipmentDetails/ShipmentDetailsMain.jsx (hunk 3 above)
Comment (WeatherfordAaron) [reply src/components/Office/ShipmentDetails/ShipmentDetailsMain.jsx:226]: Good callout. I added that test for one of the stories but didn't include testing for all the stories. I will add some testing logic to capture cases for all of the stories.

**Pull request closed without merging**